persistent/Database/Persist/TH.hs

```diff
@@ -1462,9 +1462,15 @@ fieldUpd :: Name -- ^ constructor name
     -> Exp -- ^ record value
     -> Name -- ^ field name to update
     -> Exp -- ^ new value
-    -> Exp
-fieldUpd con names record name new = CaseE record
-    [ Match (RecP con pats) (NormalB body) []]
+    -> Q Exp
+fieldUpd con names record name new = do
+    pats <-
+        fmap mconcat $ forM names $ \k -> do
+            varName <- VarP <$> newName (nameBase k)
```

Comment:
And, this line is the real fix. By creating a `newName`, we're guaranteeing that the variable is fresh and uncapturable and isn't shadowing anything.

@fumieval This is one of those tricky things about `TemplateHaskell` - it's so easy to use and make names that can cause warnings and bugs! I do it all the time and I really ought to write up a lil' how to on the matter.